### ISSUE TITLE
NaN loss for seg_loss

### ISSUE THREAD

**Issue by AmeenAli** [rulixiang/ToCo#3]
Hi

I am training with VOC and I am getting NaN value for the loss seg_loss : 

2023-03-05 15:39:27,915 - dist_train_voc_seg_neg.py - INFO: Iter: 400; Elasped: 0:17:32; ETA: 14:19:08; LR: 1.596e-05; cls_loss: 0.2528, cls_loss_aux: 0.2678, ptc_loss: 0.4553, ctc_loss: 1.0867, seg_loss: nan...

Any idea why this might happen ?

**Comment by rulixiang** [comment]
Sorry for the late reply. I'm not sure why the NaN issue happens. Maybe this [issue and reply](https://github.com/rulixiang/ToCo/issues/1#issuecomment-1459662379) can help you.

**Comment by AmeenAli** [comment]
@rulixiang  Actually I know the source of the error:

it happens in https://github.com/rulixiang/ToCo/blob/main/model/losses.py#L54

the cross-entropy loss returns nans because of the following inputs :

pred : 
tensor([[[[ 0.0318,  0.0318,  0.0318,  ...,  0.0562,  0.0562,  0.0562],
          [ 0.0318,  0.0318,  0.0318,  ...,  0.0562,  0.0562,  0.0562],
          [ 0.0318,  0.0318,  0.0318,  ...,  0.0562,  0.0562,  0.0562],
          ...,
          [-0.0057, -0.0057, -0.0057,  ...,  0.0398,  0.0398,  0.0398],
          [-0.0057, -0.0057, -0.0057,  ...,  0.0398,  0.0398,  0.0398],
          [-0.0057, -0.0057, -0.0057,  ...,  0.0398,  0.0398,  0.0398]],

         [[-0.0148, -0.0148, -0.0148,  ...,  0.0135,  0.0135,  0.0135],
          [-0.0148, -0.0148, -0.0148,  ...,  0.0135,  0.0135,  0.0135],
          [-0.0148, -0.0148, -0.0148,  ...,  0.0135,  0.0135,  0.0135],
          ...,
          [-0.0186, -0.0186, -0.0186,  ..., -0.0373, -0.0373, -0.0373],
          [-0.0186, -0.0186, -0.0186,  ..., -0.0373, -0.0373, -0.0373],
          [-0.0186, -0.0186, -0.0186,  ..., -0.0373, -0.0373, -0.0373]],

         [[ 0.0013,  0.0013,  0.0013,  ..., -0.0190, -0.0190, -0.0190],
          [ 0.0013,  0.0013,  0.0013,  ..., -0.0190, -0.0190, -0.0190],
          [ 0.0013,  0.0013,  0.0013,  ..., -0.0190, -0.0190, -0.0190],
          ...,
          [-0.0182, -0.0182, -0.0182,  ..., -0.0065, -0.0065, -0.0065],
          [-0.0182, -0.0182, -0.0182,  ..., -0.0065, -0.0065, -0.0065],
          [-0.0182, -0.0182, -0.0182,  ..., -0.0065, -0.0065, -0.0065]],

         ...,

         [[ 0.0055,  0.0055,  0.0055,  ..., -0.0611, -0.0611, -0.0611],
          [ 0.0055,  0.0055,  0.0055,  ..., -0.0611, -0.0611, -0.0611],
          [ 0.0055,  0.0055,  0.0055,  ..., -0.0611, -0.0611, -0.0611],
          ...,
          [-0.0358, -0.0358, -0.0358,  ...,  0.0399,  0.0399,  0.0399],
          [-0.0358, -0.0358, -0.0358,  ...,  0.0399,  0.0399,  0.0399],
          [-0.0358, -0.0358, -0.0358,  ...,  0.0399,  0.0399,  0.0399]],

         [[-0.0407, -0.0407, -0.0407,  ..., -0.0642, -0.0642, -0.0642],
          [-0.0407, -0.0407, -0.0407,  ..., -0.0642, -0.0642, -0.0642],
          [-0.0407, -0.0407, -0.0407,  ..., -0.0642, -0.0642, -0.0642],
          ...,
          [-0.0352, -0.0352, -0.0352,  ...,  0.0260,  0.0260,  0.0260],
          [-0.0352, -0.0352, -0.0352,  ...,  0.0260,  0.0260,  0.0260],
          [-0.0352, -0.0352, -0.0352,  ...,  0.0260,  0.0260,  0.0260]],

         [[-0.0292, -0.0292, -0.0292,  ...,  0.0016,  0.0016,  0.0016],
          [-0.0292, -0.0292, -0.0292,  ...,  0.0016,  0.0016,  0.0016],
          [-0.0292, -0.0292, -0.0292,  ...,  0.0016,  0.0016,  0.0016],
          ...,
          [-0.0400, -0.0400, -0.0400,  ..., -0.0010, -0.0010, -0.0010],
          [-0.0400, -0.0400, -0.0400,  ..., -0.0010, -0.0010, -0.0010],
          [-0.0400, -0.0400, -0.0400,  ..., -0.0010, -0.0010, -0.0010]]],


        [[[ 0.0046,  0.0046,  0.0046,  ...,  0.0664,  0.0664,  0.0664],
          [ 0.0046,  0.0046,  0.0046,  ...,  0.0664,  0.0664,  0.0664],
          [ 0.0046,  0.0046,  0.0046,  ...,  0.0664,  0.0664,  0.0664],
          ...,
          [-0.0259, -0.0259, -0.0259,  ...,  0.0343,  0.0343,  0.0343],
          [-0.0259, -0.0259, -0.0259,  ...,  0.0343,  0.0343,  0.0343],
          [-0.0259, -0.0259, -0.0259,  ...,  0.0343,  0.0343,  0.0343]],

         [[-0.0444, -0.0444, -0.0444,  ..., -0.0705, -0.0705, -0.0705],
          [-0.0444, -0.0444, -0.0444,  ..., -0.0705, -0.0705, -0.0705],
          [-0.0444, -0.0444, -0.0444,  ..., -0.0705, -0.0705, -0.0705],
          ...,
          [-0.0920, -0.0920, -0.0920,  ..., -0.0657, -0.0657, -0.0657],
          [-0.0920, -0.0920, -0.0920,  ..., -0.0657, -0.0657, -0.0657],
          [-0.0920, -0.0920, -0.0920,  ..., -0.0657, -0.0657, -0.0657]],

         [[-0.0149, -0.0149, -0.0149,  ...,  0.0044,  0.0044,  0.0044],
          [-0.0149, -0.0149, -0.0149,  ...,  0.0044,  0.0044,  0.0044],
          [-0.0149, -0.0149, -0.0149,  ...,  0.0044,  0.0044,  0.0044],
          ...,
          [-0.0122, -0.0122, -0.0122,  ...,  0.0149,  0.0149,  0.0149],
          [-0.0122, -0.0122, -0.0122,  ...,  0.0149,  0.0149,  0.0149],
          [-0.0122, -0.0122, -0.0122,  ...,  0.0149,  0.0149,  0.0149]],

         ...,

         [[-0.0038, -0.0038, -0.0038,  ..., -0.0761, -0.0761, -0.0761],
          [-0.0038, -0.0038, -0.0038,  ..., -0.0761, -0.0761, -0.0761],
          [-0.0038, -0.0038, -0.0038,  ..., -0.0761, -0.0761, -0.0761],
          ...,
          [-0.0203, -0.0203, -0.0203,  ..., -0.0343, -0.0343, -0.0343],
          [-0.0203, -0.0203, -0.0203,  ..., -0.0343, -0.0343, -0.0343],
          [-0.0203, -0.0203, -0.0203,  ..., -0.0343, -0.0343, -0.0343]],

         [[-0.1022, -0.1022, -0.1022,  ..., -0.0875, -0.0875, -0.0875],
          [-0.1022, -0.1022, -0.1022,  ..., -0.0875, -0.0875, -0.0875],
          [-0.1022, -0.1022, -0.1022,  ..., -0.0875, -0.0875, -0.0875],
          ...,
          [-0.0040, -0.0040, -0.0040,  ..., -0.0146, -0.0146, -0.0146],
          [-0.0040, -0.0040, -0.0040,  ..., -0.0146, -0.0146, -0.0146],
          [-0.0040, -0.0040, -0.0040,  ..., -0.0146, -0.0146, -0.0146]],

         [[-0.0020, -0.0020, -0.0020,  ..., -0.0467, -0.0467, -0.0467],
          [-0.0020, -0.0020, -0.0020,  ..., -0.0467, -0.0467, -0.0467],
          [-0.0020, -0.0020, -0.0020,  ..., -0.0467, -0.0467, -0.0467],
          ...,
          [ 0.0436,  0.0436,  0.0436,  ...,  0.0210,  0.0210,  0.0210],
          [ 0.0436,  0.0436,  0.0436,  ...,  0.0210,  0.0210,  0.0210],
          [ 0.0436,  0.0436,  0.0436,  ...,  0.0210,  0.0210,  0.0210]]]],

and  fg_label : 

tensor([[[255, 255, 255,  ..., 255, 255, 255],
         [255, 255, 255,  ..., 255, 255, 255],
         [255, 255, 255,  ..., 255, 255, 255],
         ...,
         [255, 255, 255,  ..., 255, 255, 255],
         [255, 255, 255,  ..., 255, 255, 255],
         [255, 255, 255,  ..., 255, 255, 255]],

        [[255, 255, 255,  ..., 255, 255, 255],
         [255, 255, 255,  ..., 255, 255, 255],
         [255, 255, 255,  ..., 255, 255, 255],
         ...,
         [255, 255, 255,  ..., 255, 255, 255],
         [255, 255, 255,  ..., 255, 255, 255],
         [255, 255, 255,  ..., 255, 255, 255]]], device='cuda:0')

any idea why that is happening ?
